packages/host/app/components/card-catalog/modal.gts

```diff
@@ -259,22 +273,31 @@ export default class CardCatalogModal extends Component<Signature> {
   }
 
   get displayedRealms(): RealmCards[] {
+    if (!this.state) {
+      return [];
+    }
     // filters the available realm cards by selected realms
     return this.state.selectedRealms.length
       ? this.state.selectedRealms
       : this.availableRealms;
   }
 
-  get state(): State {
+  get state(): State | undefined {
```

Comment:
this missing `undefined` type was crucial in addressing flaky test failures in this module--we were not acknowledging that the state could be undefined if the state stack was empty.

Reply:
👍